src/rf/apps/uploads/views.py

```diff
@@ -3,31 +3,50 @@
 from __future__ import unicode_literals
 from __future__ import division
 
-# NEEDED WHEN WE ADD PROPER AUTHORIZATION
-# from django.contrib.auth.models import User
+from django.contrib.auth.models import User
 
-from rest_framework.response import Response
-from rest_framework import decorators
-from rest_framework.permissions import AllowAny
+from rest_framework import status
+from rest_framework.exceptions import APIException
+from rest_framework.permissions import IsAuthenticated
 from rest_framework.renderers import StaticHTMLRenderer
+from rest_framework.response import Response
+from rest_framework.views import APIView
 
 from django.conf import settings
 
 import boto
 import boto.auth
 import boto.provider
+import uuid
+
+
+class SignUploadRequestView(APIView):
+    permission_classes = [IsAuthenticated]
+    renderer_classes = (StaticHTMLRenderer, )
+
+    # TODO - WE'LL WANT TO ENSURE THE USER CAN IN FACT ADD NEW CONTENT TO THE
+    # BUCKET BEFORE WE SIGN THE REQUEST.
+    def get(self, request, format=None):
+        user_id = request.user.id
+        bucket_name = settings.AWS_BUCKET_NAME + '/'
+        to_sign = str(request.REQUEST.get('to_sign'))
+
+        # Extract file name from the string to sign and then extract the user
+        # id from that.
+        start = len(bucket_name) + to_sign.find(bucket_name)
+        end = to_sign.rfind('?uploads', start)
+        file_name = to_sign[start:end]
+        sent_user_id = int(file_name[0:file_name.find('-')])
```

Comment:
I'd like to keep it. My thoughts on this:
 * The uuid can't be trusted completely. It comes from the front end. So a nasty user might be able to send us a uuid that corresponds to something that is currently up on S3. Pre-pending the user id ensures that if they do something nasty it can only mess up their own files.
 * While it is the end assumption that we'll be deleting the files, we don't know when exactly that will happen or how long they will be needed or how long the queue could get. What if a 2Gb file gets overwritten with a cat picture after an hour of time on the wire? And my feeling is that any surface area to prevent evilbad users from messing with the data should be taken particularly when we're only talking about a half dozen lines of code or so.